### ISSUE TITLE
Implicit multiplication

### ISSUE THREAD

**Issue by creativecreatorormaybenot** [fkleon/math-expressions#42]
## Problem

In mathematics, you can usually write `2x` to refer to `2 * x`. This is also how e.g. calculators, TeX, etc. handle it.

`math_expressions` does not seem to be able to handle implicit multiplication.

Using `2x` in an expression will throw an exception.

**Comment by fkleon** [comment]
Yep, this is not supported in the parser at the moment. Will have a look at this for the parser rewrite.

**Comment by creativecreatorormaybenot** [comment]
@fkleon Super exciting news! 🚀 

Let me know if I can be of any help in that regard.

**Comment by fkleon** [comment]
Support for this has been partially implemented in #87 and #88 by @juca1331. It still requires use of some parentheses  (e.g. `(2)x`) and likely doesn't work for all cases. It's opt-in via parser options. Feel free to have a play with this and leave any feedback here for improvements.
